libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -883,30 +882,49 @@ - (void)retrievedIdentityData
 
 - (void)showRetryAlertForAuthError:(NSError *)error alertTag:(NSInteger)tag
 {
-    if (nil == _statusAlert) {
-        // show alert and allow retry
-        [self log:SFLogLevelError format:@"Error during authentication: %@", error];
-        _statusAlert = [[UIAlertView alloc] initWithTitle:[SFSDKResourceUtils localizedString:kAlertErrorTitleKey]
-                                                  message:[NSString stringWithFormat:[SFSDKResourceUtils localizedString:kAlertConnectionErrorFormatStringKey], [error localizedDescription]]
-                                                 delegate:self
-                                        cancelButtonTitle:[SFSDKResourceUtils localizedString:kAlertRetryButtonKey]
-                                        otherButtonTitles: nil];
-        _statusAlert.tag = tag;
-        [_statusAlert show];
-    }
+    [self log:SFLogLevelError format:@"Error during authentication: %@", error];
+    [self showAlertWithTitle:[SFSDKResourceUtils localizedString:kAlertErrorTitleKey]
+                     message:[NSString stringWithFormat:[SFSDKResourceUtils localizedString:kAlertConnectionErrorFormatStringKey], [error localizedDescription]]
+                 buttonTitle:[SFSDKResourceUtils localizedString:kAlertRetryButtonKey]
+                         tag:tag];
 }
 
 - (void)showAlertForConnectedAppVersionMismatchError
 {
+    [self showAlertWithTitle:[SFSDKResourceUtils localizedString:kAlertErrorTitleKey]
+                     message:[SFSDKResourceUtils localizedString:kAlertVersionMismatchErrorKey]
+                 buttonTitle:[SFSDKResourceUtils localizedString:kAlertOkButtonKey]
+                         tag:kConnectedAppVersionMismatchViewTag];
+}
+
+- (void)showAlertWithTitle:(nullable NSString *)title message:(nullable NSString *)message buttonTitle:(nullable NSString *)buttonTitle tag:(NSInteger)tag
+{
+
     if (nil == _statusAlert) {
-        // Show alert and execute failure block.
-        _statusAlert = [[UIAlertView alloc] initWithTitle:[SFSDKResourceUtils localizedString:kAlertErrorTitleKey]
-                                                  message:[SFSDKResourceUtils localizedString:kAlertVersionMismatchErrorKey]
-                                                 delegate:self
-                                        cancelButtonTitle:[SFSDKResourceUtils localizedString:kAlertOkButtonKey]
-                                        otherButtonTitles: nil];
-        _statusAlert.tag = kConnectedAppVersionMismatchViewTag;
-        [_statusAlert show];
+        __weak SFAuthenticationManager *weakSelf = self;
+        _statusAlert = [UIAlertController alertControllerWithTitle:title
```

Comment:
Oh thanks.  I mostly meant for this property specifically, but thanks for covering all of the bases!